tap_jira/__init__.py

```diff
@@ -109,18 +114,22 @@ def sync():
 def main():
     args = get_args()
 
+    jira_config = args.config
+    # jira client instance
+    jira_client = Client(jira_config)
+
     # Setup Context
     catalog = Catalog.from_dict(args.properties) \
-        if args.properties else discover()
-    Context.config = args.config
+        if args.properties else discover(jira_client) # Pass jira_client instance to switch pk for users stream in the catalog
+    Context.config = jira_config
     Context.state = args.state
     Context.catalog = catalog
 
-    Context.client = Client(Context.config)
-
+    Context.client = jira_client
     try:
         if args.discover:
-            discover().dump()
+            # Pass jira_client instance to switch pk for users stream in the catalog
+            discover(jira_client).dump()
```

Comment:
Define `Context.client` at starting of the `main()` function and use it directly where it is needed in discover mode rather than passing `jira_client` everywhere. Take reference from sync mode workflow as it didn't pass client object in any functions.

Reply:
Updated